cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/task/ProvisionTask.java

```diff
@@ -108,6 +109,10 @@ protected void handleStateSaveFailure(ProvisioningTaskInfo taskInfo, Exception e
 
   private ProvisioningSubtask createClusterCreateSubtask() {
     return new ClusterCreateSubtask(provisioner, provisionerContext, cluster -> {
+      if (!Objects.equals(provisionerContext.getCDAPVersion(), provisionerContext.getAppCDAPVersionInfo())) {
+        LOG.info("Starting a pipeline created with a previous CDAP version. " +
+                   "Please consider upgrading a pipeline to employ all the enhancements");
```

Comment:
nit: upgrading a pipeline -> upgrading the pipeline

Reply:
done